nomad/leader.go

```diff
@@ -625,6 +627,29 @@ func (s *Server) publishJobSummaryMetrics(stopCh chan struct{}) {
 								Value: name,
 							},
 						}
+
+						if strings.Contains(summary.JobID, "/dispatch-") {
+							jobInfo := strings.Split(summary.JobID, "/dispatch-")
+							labels = append(labels, metrics.Label{
+								Name:  "parent_id",
+								Value: jobInfo[0],
+							}, metrics.Label{
+								Name:  "dispatch_id",
+								Value: jobInfo[1],
+							})
+						}
+
+						if strings.Contains(summary.JobID, "/periodic-") {
+							jobInfo := strings.Split(summary.JobID, "/periodic-")
+							labels = append(labels, metrics.Label{
+								Name:  "parent_id",
+								Value: jobInfo[0],
+							}, metrics.Label{
+								Name:  "periodic_id",
+								Value: jobInfo[1],
```

Comment:
Ensure that the length of `jobInfo` is what is expected for safety.

Reply:
In this specific case, the contains check in line 642 guards against that. If the string ends with `/periodic-` that slice should still have 2 elements with one being the empty string.